api/models/model.py

```diff
@@ -119,7 +120,9 @@ def getByTid(self, tid):
     def getUserModelsByUid(self, uid, is_current_user=False, n=5, offset=0):
         query_res = self.dbs.query(Model).filter(Model.uid == uid)
         if not is_current_user:
-            query_res = query_res.filter(Model.is_published == True)  # noqa
+            query_res = query_res.filter(
+                db.and_(Model.is_published == True, Model.is_anonymous == False)
```

Comment:
weirdly, I couldn't find any instance where `getUserModelsByUidAndMid` is actually used 

it's called in `get_user_model` in `tasks.py`. That has an endpoint `/users/<uid:int>/models/<model_id:int>`... which is never actually fetched in any function in ApiService.js

based on the endpoint it seems like it would be used when choosing a specific model from a user's `ModelSubPage`, but that just calls the model's actual `ModelPage` (for the better IMO)

![where_api_call_lol](https://user-images.githubusercontent.com/29654756/137413931-1fab7946-3bdb-4e1a-86cb-c3a6662a9fc9.gif)

Reply:
Thanks for looking into this. The fact that the endpoint exists means that an adversary could talk to it to get the name of the person who uploaded an anonymous model. It could still be good to include this is_anonymous filter to any endpoints that have this vulnerability.